compiler/modules/front/resolveTypes.bal

```diff
@@ -28,6 +28,20 @@ function resolveTypes(ModuleSymbols mod) returns ResolveTypeError? {
     foreach var defn in mod.defns {
         check resolveDefn(mod, defn);
     }
+    if !mod.tc.env.isReady() {
+        // This should never happen
+        panic err:impossible("type environment is not ready");
+    }
+    foreach var { semType, modDefn, startPos, endPos } in mod.deferredEmptinessChecks {
+        check testTypeForFiniteShape(mod.tc, semType, modDefn, startPos, endPos);
+    }
+}
+
+function testTypeForFiniteShape(t:Context tc, t:SemType semType, s:ModuleLevelDefn modDefn, Position startPos, Position endPos) returns ResolveTypeError? {
```

Comment:
We are using this for the intersection case also, which might be empty without having infinite types.